sdl2promela/stop_condition/translator.py

```diff
@@ -278,36 +291,165 @@ def _generate(context: GenerateContext, expr: model.VariableReference):
     It means that variable is a selector of CHOICE.
     """
     if context.choice_selection is not None:
+        # The list of available alternatives of CHOICE is case-sensitive
+        # Stop Condition Language is case-insensitive,
+        # so, the existing identifier should be generated.
+        selected_alternatives = [
+            alternative
+            for alternative in context.choice_selection_alternatives
+            if alternative.lower() == expr.name.lower()
+        ]
+        if len(selected_alternatives) != 1:
+            raise TranslateException(
+                f"The CHOICE datatype '{context.choice_selection}' has no alternative with name '{expr.name}'"
```

Comment:
What if there are 2 alternatives?

Reply:
a Dict with two same keys? It is impossible.

Reply:
Is it impossible though given different letter casing? The comment states "The list of available alternatives of CHOICE is case-sensitive" which to me suggests that there can be This, and thIS, and THIS and this.
Also, if you are sure, please comment why (is it ensured by opengeode? is error reported during parsing?).

Reply:
Ok, so I have no idea if it is possible to declare two alternatives in CHOICE that have the same name but different case (asn1scc crashes, bug reported), However it is possible to declare such SEQUENCE. The issue occurs later, the Ada generated by asn1scc or Ada generated by OpenGEODE won't compile at all - Ada is a case-insensitive language.
So, IMO we shouldn't support this scenario, it causes an error in TASTE.

From the other side, promela is a case sensitive language and the definitions generated in `dataview.pml` have some name convention originated from asn1scc. The stop condition language is case insensitive, which means that `sdl2promela` should ensure that in the final code the generated identifiers will be valid.

Reply:
Please split it into if len(selected_alternatives) == 0 and if len(selected_alternatives) > 1, with appropriate error messages